File: processor/src/test/resources/approvals/beanvalidation/OrderDateValueExtractor.java
package test;

import jakarta.validation.valueextraction.ExtractedValue;
import jakarta.validation.valueextraction.ValueExtractor;
import scenarios.java.OrderDate;

import javax.annotation.processing.Generated;
import java.time.LocalDate;

@Generated("com.qualityminds.lazyval.processor.internal.codegen.beanvalidation.BeanValidationGenerator")
public class OrderDateValueExtractor implements ValueExtractor<@ExtractedValue(type = LocalDate.class) OrderDate> {
  @Override
  public void extractValues(OrderDate originalValue, ValueExtractor.ValueReceiver receiver) {
    if (originalValue == null) {
      receiver.value(null, null);
      return;
    }
    receiver.value(null, originalValue.value());
  }
}

File: ksp/src/test/resources/approvals/beanvalidation/OrderDateValueExtractor.kt
package test.domain

import jakarta.`annotation`.Generated
import jakarta.validation.valueextraction.ValueExtractor
import jakarta.validation.valueextraction.UnwrapByDefault
import scenarios.kotlin.OrderDate

@Generated("com.qualityminds.lazyval.ksp.internal.codegen.beanvalidation.BeanValidationGenerator")
@UnwrapByDefault
public class OrderDateValueExtractor : OrderDateValueExtractorBase() {
  override fun extractValues(originalValue: OrderDate?, `receiver`: ValueExtractor.ValueReceiver) {
    if (originalValue == null) {
      receiver.value(null, null)
      return
    }
    receiver.value(null, originalValue.date)
  }
}
